app/code/Magento/Bundle/Model/LinkManagement.php

```diff
@@ -239,6 +241,55 @@ protected function mapProductLinkToSelectionModel(
         return $selectionModel;
     }
 
+    /**
+     * Fill selection model with product link data.
+     *
+     * @param Selection $selectionModel
+     * @param LinkInterface $productLink
+     * @param ProductInterface $parentProduct
+     * @param int $linkedProductId
+     * @param string $linkField
+     * @return Selection
+     * @throws NoSuchEntityException
+     */
+    private function mapProductLinkToBundleSelectionModel(
+        Selection $selectionModel,
+        LinkInterface $productLink,
+        ProductInterface $parentProduct,
+        int $linkedProductId,
+        string $linkField
+    ): Selection {
+        $selectionModel->setProductId($linkedProductId);
+        $selectionModel->setParentProductId($parentProduct->getData($linkField));
+        if ($productLink->getSelectionId() !== null) {
+            $selectionModel->setSelectionId($productLink->getSelectionId());
+        }
+        if ($productLink->getOptionId() !== null) {
+            $selectionModel->setOptionId($productLink->getOptionId());
+        }
+        if ($productLink->getPosition() !== null) {
+            $selectionModel->setPosition($productLink->getPosition());
+        }
+        if ($productLink->getQty() !== null) {
+            $selectionModel->setSelectionQty($productLink->getQty());
+        }
+        if ($productLink->getPriceType() !== null) {
+            $selectionModel->setSelectionPriceType($productLink->getPriceType());
+        }
+        if ($productLink->getPrice() !== null) {
+            $selectionModel->setSelectionPriceValue($productLink->getPrice());
+        }
+        if ($productLink->getCanChangeQuantity() !== null) {
+            $selectionModel->setSelectionCanChangeQty($productLink->getCanChangeQuantity());
+        }
+        if ($productLink->getIsDefault() !== null) {
+            $selectionModel->setIsDefault($productLink->getIsDefault());
+        }
+        $selectionModel->setWebsiteId((int)$this->storeManager->getStore($parentProduct->getStoreId())->getWebsiteId());
```

Comment:
As I can see when we load the product on line 322 we are not using the `store_id`, in case of a product being assigned to multiple stores, would such approach still work?